lib/screens/edit_product_screen.dart

```diff
@@ -8,18 +9,161 @@ class EditProductScreen extends StatefulWidget {
 }
 
 class _EditProductScreenState extends State<EditProductScreen> {
+  final _priceFocusNode = FocusNode();
+  final _descriptionFocusNode = FocusNode();
+  final _imageUrlController = TextEditingController();
+  final _imageUrlFocusNode = FocusNode();
+  final _form = GlobalKey<FormState>();
+  var _editedProduct = Product(
+    id: null,
+    title: '',
+    price: 0,
+    description: '',
+    imageUrl: '',
+  );
+
+  @override
+  void initState() {
+    _imageUrlFocusNode.addListener(_updateImageUrl);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _imageUrlFocusNode.removeListener(_updateImageUrl);
+    _priceFocusNode.dispose();
+    _descriptionFocusNode.dispose();
+    _imageUrlController.dispose();
+    _imageUrlFocusNode.dispose();
+    super.dispose();
+  }
+
+  void _updateImageUrl() {
+    if (!_imageUrlFocusNode.hasFocus) {
+      setState(() {});
+    }
+  }
+
+  void _saveForm() => _form.currentState.save();
+
+  Product buildProductImageUrl(Product eProduct, String value) {
+    return Product(
+      title: eProduct.title,
+      price: eProduct.price,
+      description: eProduct.description,
+      imageUrl: value,
+      id: null,
+    );
+  }
+
+  Product buildProductDescription(Product eProduct, String value) {
+    return Product(
+      title: eProduct.title,
+      price: eProduct.price,
+      description: value,
+      imageUrl: eProduct.imageUrl,
+      id: null,
+    );
+  }
+
+  Product buildProductPrice(Product eProduct, String value) {
+    return Product(
+      title: eProduct.title,
+      price: double.parse(value),
+      description: eProduct.description,
+      imageUrl: eProduct.imageUrl,
+      id: null,
+    );
+  }
+
+  Product buildProductTitle(String value, Product eProduct) {
+    return Product(
+      title: value,
+      price: eProduct.price,
+      description: eProduct.description,
+      imageUrl: eProduct.imageUrl,
+      id: null,
+    );
+  }
```

Comment:
Acceptable for now, but this can be improved if this dependency is used:

https://pub.dev/packages/freezed

Reply:
will take note of this and implement it after lessons in section 9